### PR TITLE
Subscription audio notification

### LINKED ISSUE #71: Improve sound of new direct message
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/19161066-improve-sound-of-new-direct-message?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #171: Option to enable/disable notification sound/message
The notification sound (which appears to be taken from Facebook?) gets quite annoying after a while because every time I hear it I instinctively check Facebook. It'd be nice to be able to either turn it off entirely, or set which channels play a notification sound. Additionally, being able to choose another notification sound (or you switching away from the Facebook one entirely) would probably be good, to prevent the instinctive Facebook checking that I mentioned above.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/20884136-option-to-enable-disable-notification-sound-message?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by marceloschmidt**
@RocketChat/core 

Closes #171
Closes #71 
Related to #1268 

image

image

image

### CODE REVIEW

**Comment by rodrigok** [comment]
Translation is missing
captura de tela 2017-01-03 as 17 32 31

The custom sound per room is not working, it is getting the sound of the current opened room for all notifications and not the sound for the room of the notification

**Comment by marceloschmidt** [comment]
Translation is missing in Portuguese and in many other languages... translators do the work after our merges...

**Comment by marceloschmidt** [comment]
I have fixed the issue with sounds being played based on currently opened room. Please review again @rodrigok ? Thanks.

**Comment by engelgabriel** [comment]
@marceloschmidt you need to resolve the conflicts now

**Comment by rodrigok** [comment]
@marceloschmidt the admin menu `Custom Sounds` is not working:
image

**Comment by rodrigok** [comment]
@marceloschmidt Fixed 😃

File: packages/rocketchat-custom-sounds/admin/adminSounds.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/* globals isSetNotNull, RocketChatTabBar */
+Template.adminSounds.helpers({
+	isReady() {
+		if (isSetNotNull(() => Template.instance().ready)) {
```
Comment (rodrigok) [inline packages/rocketchat-custom-sounds/admin/adminSounds.js]: What is the reason for use this `isSetNotNull`?

File: packages/rocketchat-custom-sounds/admin/adminSounds.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+Template.adminSounds.onRendered(() =>
+	Tracker.afterFlush(function() {
+		SideNav.setFlex('adminFlex');
+		SideNav.openFlex();
+	})
+);
+
+Template.adminSounds.events({
+	['keydown #sound-filter'](e) {
```
Comment (rodrigok) [inline packages/rocketchat-custom-sounds/admin/adminSounds.js]: Why the selector is between brackets?

File: packages/rocketchat-custom-sounds/admin/adminSounds.js (hunk 1 above)
Comment (marceloschmidt) [inline packages/rocketchat-custom-sounds/admin/adminSounds.js]: CTRL+C, CTRL+V from adminEmoji :( should I remove it?

File: packages/rocketchat-custom-sounds/admin/adminSounds.js (hunk 2 above)
Comment (marceloschmidt) [inline packages/rocketchat-custom-sounds/admin/adminSounds.js]: Likewise above

**Pull request merged**